tests/FlashTest.php

```diff
@@ -48,4 +49,39 @@ public function testInstances()
         $this->assertEquals('value2', $flash2->get('key'));
         $this->assertEquals('value3', $flash3->get('key'));
     }
+
+    public function testNextSession()
+    {
+        $flash = Flash::getInstance();
+        $flash->set('key', 'value');
+        $this->nextPageLoad();
+        $flash = Flash::getInstance();
+        $this->assertEquals('value', $flash->get('key'));
+        $this->nextPageLoad();
+        $flash = Flash::getInstance();
+        $this->assertNull($flash->get('key'));
+    }
+
+    public function testOverlappingSessions()
+    {
+        $flash = Flash::getInstance();
+        $flash->set('key', 'value');
+        $this->nextPageLoad();
+        $flash = Flash::getInstance();
+        $this->assertEquals('value', $flash->get('key'));
+        $flash->set('key2', 'value2');
+        $this->nextPageLoad();
+        $flash = Flash::getInstance();
+        $this->assertNull($flash->get('key'));
```

Comment:
Maybe check if `key2` is still there?
```suggestion
        $this->assertNull($flash->get('key'));
        $this->assertEquals('value2', $flash->get('key2'));
```